deepinv/physics/noise.py

```diff
@@ -90,6 +97,82 @@ def randn_like(self, input: torch.Tensor, seed: int = None):
         self.rng_manual_seed(seed)
         return torch.empty_like(input).normal_(generator=self.rng)
 
+    def update_parameters(self, **kwargs):
+        r"""
+
+        Update the parameters of the noise model.
+
+        :param dict kwargs: dictionary of parameters to update.
+        """
+        if kwargs:
+            for key, value in kwargs.items():
+                if (
+                    value is not None
+                    and hasattr(self, key)
+                    and isinstance(value, torch.Tensor)
+                ):
+                    self.register_buffer(key, value)
+
+    def _float_to_tensor(self, value):
+        if value is None:
+            return value
+        elif isinstance(value, (float, int)):
+            return torch.tensor(value, dtype=torch.float32)
+        elif isinstance(value, torch.Tensor):
+            return value
+        else:
+            raise ValueError
+
+    # To handle the transfer between CPU/GPU properly
+    def to(self, *args, **kwargs):
+        super().to(*args, **kwargs)
+        device = self._get_device_from_args(*args, **kwargs)
+        if device is not None and self.rng is not None:
+            state = self.rng.get_state()
+            # Move the generator to the specified device
+            self.rng = torch.Generator(device=device)
+            try:
+                self.rng.set_state(state)
+            except RuntimeError:
+                warnings.warn(
+                    "Moving the random number generator between CPU/GPU is not possible. Re-initializing the generator."
```

Comment:
```suggestion
                    "Moving the random number generator between CPU/GPU is not possible. Reinitialize the generator on the correct device."
```

how often does this happen?

Reply:
I only happens if one initialize the rng on CPU and send to GPU